src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -301,68 +317,61 @@ private static string DecodePropertyName(string name, string language)
-            {
-                _builder = builder;
-                _generator = new Generator(builder);
-            }
+            private readonly StringBuilder _builder = builder;
+            private readonly DeclarationGenerator _generator = new DeclarationGenerator(builder);
 
-            public override void DefaultVisit(ISymbol symbol)
-            {
-                throw new InvalidOperationException("Cannot generated a documentation comment id for symbol.");
-            }
+            /// <summary>
+            /// If we hit anything we don't know about, indicate failure.
+            /// </summary>
+            public override bool DefaultVisit(ISymbol symbol)
+                => false;
 
-            public override void VisitEvent(IEventSymbol symbol)
+            public override bool VisitEvent(IEventSymbol symbol)
             {
                 _builder.Append("E:");
-                _generator.Visit(symbol);
+                return _generator.Visit(symbol);
             }
 
-            public override void VisitField(IFieldSymbol symbol)
+            public override bool VisitField(IFieldSymbol symbol)
             {
                 _builder.Append("F:");
-                _generator.Visit(symbol);
+                return _generator.Visit(symbol);
             }
 
-            public override void VisitProperty(IPropertySymbol symbol)
+            public override bool VisitProperty(IPropertySymbol symbol)
             {
                 _builder.Append("P:");
-                _generator.Visit(symbol);
+                return _generator.Visit(symbol);
             }
 
-            public override void VisitMethod(IMethodSymbol symbol)
+            public override bool VisitMethod(IMethodSymbol symbol)
             {
                 _builder.Append("M:");
-                _generator.Visit(symbol);
+                return _generator.Visit(symbol);
             }
 
-            public override void VisitNamespace(INamespaceSymbol symbol)
+            public override bool VisitNamespace(INamespaceSymbol symbol)
             {
                 _builder.Append("N:");
-                _generator.Visit(symbol);
+                return _generator.Visit(symbol);
             }
 
-            public override void VisitNamedType(INamedTypeSymbol symbol)
+            public override bool VisitNamedType(INamedTypeSymbol symbol)
             {
                 _builder.Append("T:");
-                _generator.Visit(symbol);
+                return _generator.Visit(symbol);
             }
 
-            private class Generator : SymbolVisitor<bool>
+            private sealed class DeclarationGenerator(StringBuilder builder) : SymbolVisitor<bool>
             {
-                private readonly StringBuilder _builder;
+                private readonly StringBuilder _builder = builder;
                 private ReferenceGenerator? _referenceGenerator;
 
-                public Generator(StringBuilder builder)
```

Comment:
>public Generator(StringBuilder builder) [](http://example.com/codeflow?start=16&length=39)

Please go back to the regular constructor

Reply:
Moved to cleanup pr.